### PR TITLE
LPS-180230 [Raylife AP] Refactoring policies details navigator

### PR DESCRIPTION
**Pull request by eduardo-diniz**
[LPS-180230](https://issues.liferay.com/browse/LPS-180230)

### CODE REVIEW

**Comment by eduardo-diniz** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/driver-info/index.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+function calculatedAge(dateOfBirth: string) {
+	return Math.floor(
+		Math.ceil(
+			Math.abs(Date.parse(dateOfBirth) - Date.now()) / (1000 * 3600 * 24)
+		) / 365.25
+	);
+}
+
+type ApplicationPolicyDetailsType = {
+	dataJSON: any;
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/driver-info/index.tsx:28]: Please type it correctly.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/driver-info/index.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+import '../index.scss';
+
+function calculatedAge(dateOfBirth: string) {
+	return Math.floor(
+		Math.ceil(
+			Math.abs(Date.parse(dateOfBirth) - Date.now()) / (1000 * 3600 * 24)
+		) / 365.25
+	);
+}
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/driver-info/index.tsx:25]: Please move it to the "utils" folder.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/history-info/index.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+import {InfoPanelType} from '..';
+
+import PanelComponent from '../../../../../../common/components/panel';
+
+type HistoryInfoType = {
+	arraySortedByDate: any;
+	setShowPanel: any;
+	showPanel: any;
+};
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/history-info/index.tsx:23]: Please type it.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-summary-details/index.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import Summary from '../../../../../common/components/summary';
+
+const PolicySummary = ({application, policy}: any) => {
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-summary-details/index.tsx:17]: Please type it.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/index.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	email,
+	phone,
+}: ApplicationPolicyDetailsType) => {
+	const navbarLabel = [
+		NavBarLabel.Vehicles,
+		NavBarLabel.Drivers,
+		NavBarLabel.History,
+	];
+	const [active, setActive] = useState(navbarLabel[0]);
+	const [applicationData, setApplicationData] = useState<any>();
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/index.tsx:68]: Please type it.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/vehicle-info/index.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+ */
+
+import {PolicyDetailsType} from '..';
+
+import addImageFallback from '../../../../../../common/utils/addImageFallback';
+import {getWebDavUrl} from '../../../../../../common/utils/webdav';
+
+import '../index.scss';
+
+const VehicleInfo = (applicationDataJSON: any) => {
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/vehicle-info/index.tsx:22]: Please type it.

**Review by joanagrego — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/driver-info/index.tsx (hunk 2 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/driver-info/index.tsx:25]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/history-info/index.tsx (hunk 3 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/PolicyDetails/policy-navigator-details/history-info/index.tsx:23]: Done!

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by joanagrego** [comment]
ci:test:relevant

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by joanagrego** [comment]
ci:forward

**Pull request closed without merging**